### PR TITLE
[Frontend] Avoid crashing in trace emission due to "duplicate" swiftinterfaces

### PR DESCRIPTION
**Pull request by varungandhi-apple**
If the build system ends up using the standard library with an `arm.swiftinterface` and an `armv7s.swiftinterface`, we will only load the latter but that should be fine (due to ABI compatibility) and we shouldn't crash in trace emission on seeing a swiftinterface in the depTracker that was not loaded.

### CODE REVIEW

**Review by harlanhaskins — APPROVED** [review]

**Review by harlanhaskins — APPROVED** [review]

File: lib/FrontendTool/FrontendTool.cpp [hunk 1]
```diff
@@ -359,15 +359,15 @@ static void computeSwiftModuleTraceInfo(
     // FIXME: The behavior of fs::exists for relative paths is undocumented.
     // Use something else instead?
     if (fs::exists(moduleAdjacentInterfacePath)) {
-      err << "Found swiftinterface at\n" << moduleAdjacentInterfacePath
-          << "\nbut it was not recorded\n";
-      errorUnexpectedPath(err);
+      // This should be an error but it is not because of funkiness around
+      // compatible modules such as us having both armv7s.swiftinterface
+      // and armv7.swiftinterface in the dependency tracker.
+      continue;
```
Comment (jrose-apple) [inline lib/FrontendTool/FrontendTool.cpp]: This one seems like you could still catch it, no? I wouldn't expect a case where you have a dependency on a swiftmodule and not a swiftinterface, even in the funky armv7s/armv7 case we talked about yesterday.

Comment (harlanhaskins) [reply lib/FrontendTool/FrontendTool.cpp]: The specific case that caught this error would have caught this error if the transitive import emits a .swiftinterface, and the second error if it doesn't emit a .swiftinterface. Just to be safe, for now, we're gonna disable both errors.

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
@swift-ci please test Windows platform

**Comment by harlanhaskins** [comment]
Smoke test passed on macOS and Linux

@swift-ci please test Windows platform

File: lib/FrontendTool/FrontendTool.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/FrontendTool/FrontendTool.cpp]: Mmk, not going to argue against safety.

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test Windows platform

**Pull request merged**